actionners/kubernetes/sysdig/sysdig.go

```diff
@@ -0,0 +1,292 @@
+  verbs:
+  - get
+  - list
+  - create
+`
+	Example string = `- action: Create a syscall capture from a pod
+  actionner: kubernetes:sysdig
+  parameters:
+    duration: 10
+    buffer_size: 1024
+  output:
+    target: aws:s3
+    parameters:
+      bucket: my-bucket
+      prefix: /captures/
+`
+)
+
+var (
+	RequiredOutputFields = []string{"k8s.ns.name, k8s.pod.name", "ka.target.namespace, (ka.target.pod.name or ka.target.name)"}
+)
+
+type Parameters struct {
+	Image      string `mapstructure:"image"`
+	Scope      string `mapstructure:"scope" validate:"oneof=pod node"`
+	Duration   int    `mapstructure:"duration" validate:"gte=0,lte=30"`
+	BufferSize int    `mapstructure:"buffer_size" validate:"gte=128"`
+}
+
+const (
+	baseName     string = "falco-talon-sysdig-"
+	defaultImage string = "issif/sysdig:latest"
+	// defaultImage       string = "sysdig/sysdig:latest"
+	defaultScope       string = "pod"
+	defaultTTL         int    = 60
+	defaultDuration    int    = 5
+	defaultMaxDuration int    = 30
+	defaultBufferSize  int    = 2048
+)
+
+type Actionner struct{}
+
+func Register() *Actionner {
+	return new(Actionner)
+}
+
+func (a Actionner) Init() error {
+	return k8s.Init()
+}
+
+func (a Actionner) Information() models.Information {
+	return models.Information{
+		Name:                 Name,
+		FullName:             Category + ":" + Name,
+		Category:             Category,
+		Description:          Description,
+		Source:               Source,
+		RequiredOutputFields: RequiredOutputFields,
+		Permissions:          Permissions,
+		Example:              Example,
+		Continue:             Continue,
+		AllowOutput:          AllowOutput,
+		RequireOutput:        RequireOutput,
+	}
+}
+
+func (a Actionner) Parameters() models.Parameters {
+	return Parameters{
+		Duration:   defaultDuration,
+		Scope:      defaultScope,
+		BufferSize: defaultBufferSize,
+		Image:      defaultImage,
+	}
+}
+
+func (a Actionner) Checks(event *events.Event, _ *rules.Action) error {
+	if event.Source == "k8s_audit" {
+		event.OutputFields["k8s.ns.name"] = event.OutputFields["ka.target.namespace"]
+		if event.OutputFields["ka.target.pod.name"] != "" {
+			event.OutputFields["k8s.pod.name"] = event.OutputFields["ka.target.pod.name"]
+		} else {
+			event.OutputFields["k8s.pod.name"] = event.OutputFields["ka.target.name"]
+		}
+	}
+	return k8sChecks.CheckPodExist(event)
+}
+
+func (a Actionner) Run(event *events.Event, action *rules.Action) (utils.LogLine, *models.Data, error) {
+	podName := event.GetPodName()
+	namespace := event.GetNamespaceName()
+
+	objects := map[string]string{
+		"pod":       podName,
+		"namespace": namespace,
+	}
+
+	var parameters Parameters
+	err := utils.DecodeParams(action.GetParameters(), &parameters)
+	if err != nil {
+		return utils.LogLine{
+			Objects: nil,
+			Error:   err.Error(),
+			Status:  utils.FailureStr,
+		}, nil, err
+	}
+
+	if parameters.Duration == 0 {
+		parameters.Duration = defaultDuration
+	}
+
+	if parameters.Duration > 30 {
```

Comment:
other than this check, I believe the other ones are already being handled by the parameter decoding from above? This one, though, seems to be required to guarantee we won't be running longer than the 30s, even if the user sets it. Would be nice to have a warning message informing we're overriding what they're passing.

Reply:
the validator is enough you're right.